### PR TITLE
Adding Silicon analyser to OSIRIS

### LINKED ISSUE #34301: New/updated reduction for Si analyser OSIRIS data
the new analyser will improve the time resolution from ps to ns. So we will need some new reduction. It is currently unclear if that is an update/extension to the current one or a new reduction.

### PR DESCRIPTION
**Pull request by AnthonyLim23**
This PR add a dummy set of values for the new Si analyser on OSIRIS. We are waiting for the true detector positions etc. Once we have received them we will update the IDF and add some tests (also need real data) https://github.com/mantidproject/mantid/issues/35777

Report to Spencer and Franz
**Summary of work**

**Further detail of work**

**To test:**

Run the below script
```

ws = Load(r'\\isis.cclrc.ac.uk\inst$\ndxosiris\instrument\data\cycle_12_5\OSIRIS00100320.nxs')


for j in range(5, 10):
    y = ws.readY(j)
    e = ws.readE(j)
    ws.setY(j+1000, y)
    ws.setE(j+1000, e)
    
SaveNexus(ws, "OSIRIS_test.nxs")
    
ISISIndirectEnergyTransferWrapper(InputFiles='OSIRIS_test.nxs',
    Instrument='OSIRIS', Analyser='graphite', Reflection='002', 
    SpectraRange='963,1004', FoldMultipleFrames=False, GroupingMethod='Individual', 
    OutputWorkspace='osiris100320_graphite111_Reduced')


ISISIndirectEnergyTransfer(InputFiles='OSIRIS_test.nxs', 
    Instrument='OSIRIS', Analyser='silicon', Reflection='111', 
    SpectraRange='1005,1010',
    OutputWorkspace='osiris100320_silicon111_Reduced')

ISISIndirectEnergyTransfer(InputFiles='OSIRIS_test.nxs', 
    Instrument='OSIRIS', Analyser='silicon', Reflection='333', 
    SpectraRange='1005,1010',
    OutputWorkspace='osiris100320_silicon333_Reduced')
```

For graphite you will see a nice peak at about 0
For 111 you will see the values rapidly increasing from about 0.5
For 333 you will see stuff with negative x values

image

Fixes #34301 . 

No release notes as we are adding this so we have something for using in the GUI upgrades. It is not ready for users (need the real detectors) 

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: instrument/OSIRIS_Definition.xml [hunk 1]
```diff
@@ -760,6 +760,32 @@
   <algebra val="app-shape" />
 </type>
 
+  <component type="silica" idlist="silica">
```
Comment (MialLewis) [inline instrument/OSIRIS_Definition.xml:763]: Can you confirm that this should indeed be `silica` in this file, not `silicon`?

**Review by MialLewis — CHANGES_REQUESTED** [review]
LGTM, just one query.

File: instrument/OSIRIS_Definition.xml (hunk 1 above)
Comment (SpencerHowells) [reply instrument/OSIRIS_Definition.xml:763]: @MialLewis @AnthonyLim23 it should be silicon

Comment (AnthonyLim23) [inline instrument/OSIRIS_Definition.xml:763]:
```suggestion
  <component type="silicon" idlist="silicon">
```

File: instrument/OSIRIS_Definition.xml [hunk 2]
```diff
@@ -760,6 +760,32 @@
...
+    <properties />
+    <location  />
+  </component>
+
+  <type name="silica">
```
Comment (AnthonyLim23) [inline instrument/OSIRIS_Definition.xml:768]:
```suggestion
  <type name="silicon">
```

File: instrument/OSIRIS_Definition.xml [hunk 3]
```diff
@@ -771,5 +797,7 @@
 <idlist idname="graphite">
   <id start="963" end="1004" />
 </idlist>
-
+  <idlist idname="silica">
```
Comment (AnthonyLim23) [inline instrument/OSIRIS_Definition.xml:800]:
```suggestion
  <idlist idname="silicon">
```

**Comment by AnthonyLim23** [comment]
unrelated build failure

**Review by MialLewis — APPROVED** [review]

**Comment by gemmaguest** [comment]
When this is merged the IDF and IPF changes will take immediate effect but the code changes will not (except in the nightly). Just checking that this is intentional and the old code will still work with the new IDF/IPF?

**Comment by AnthonyLim23** [comment]
The old code will never use the new IPF (since it is explicitly for Si).

The IDF will be ok as long as users dont try to reduce data from the spectra at the end 1005-1010. At present these spectra cannot be reduced as they are empty (0's) and have no associated mode.

**Comment by AnthonyLim23** [comment]
I think the algorithm would fall over, but it does now.

**Pull request merged**